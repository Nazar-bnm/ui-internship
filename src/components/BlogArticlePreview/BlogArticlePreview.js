import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import HttpService from '../../service/HttpService/httpService';

import './BlogArticlePreview.scss';

export const CN = 'blog';

class BlogArticlePreview extends Component {
  constructor(props) {
    super(props);

    this.state = {
      blogsList: []
    };
  }

  componentDidMount() {
    this.getBlogs();
  }

  async getBlogs() {
    const userAPI = new HttpService();

    try {
      const response = await userAPI.get(`${process.env.SERVER_URL}/blogs`);
      if (response && response.data) {
        const blogsToRender = [response.data[0], response.data[1]];
        this.setState({ blogsList: blogsToRender });
      }
    } catch (error) {
      throw new Error(error);
    }
  }

  renderBrands(blogsList) {
    return blogsList.map((item) => {
      const {
        photo,
        title,
        description
      } = item;
      const maxSymbolsNumber = 50;
      const shortText = `${description.slice(0, maxSymbolsNumber)}...`;
      const imageSrc = `${process.env.BLOG_IMAGE_URL}/${photo}`;

      return (
        <div key={title} className={`${CN}`}>
          <img
            className={`${CN}__img`}
            alt={title}
            src={imageSrc}
          />
          <div className={`${CN}__text`}>
            <h2 className={`${CN}__title`}>{title}</h2>
            <p className={`${CN}__description`}>{shortText}</p>
          </div>
        </div>
      );
    });
  }

  render() {
    const { blogsList } = this.state;
    const { className } = this.props;

    return (
      <div className={cx(CN, className)}>
        <h3 className={`${CN}__heading`}>from our blog</h3>
        <div className="blog__wrapper">
          {this.renderBrands(blogsList)}
        </div>
        <Link to="/blog" className={`${CN}__read-more`}>
          read more
        </Link>
      </div>
    );
  }
}

BlogArticlePreview.propTypes = {
  className: PropTypes.string
};

BlogArticlePreview.defaultProps = {
  className: ''
};
export default BlogArticlePreview;
