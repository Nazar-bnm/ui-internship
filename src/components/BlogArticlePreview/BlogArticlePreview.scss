@import '../../scss/global_styles';

.blog {
  position: relative;
  min-height: 200px;
  padding: 20px 0px;

  &__heading {
    @include title-font;
  }

  &__wrapper {
    display: block;
  }

  &:nth-of-type(2) {
    display: none;
  }

  &__text {
    position: absolute;
    bottom: 0;
    right: 0;
    width: 70%;
    padding: 10px;
    background-color:$white;
    text-transform: uppercase;
  }

  &__title {
    @include title-font;
    margin-bottom: 5px;
    font-size: 1rem;
    text-align: left;
  }

  &__description {
    font-family: font(primary-light);
    font-size: 0.7rem;
    color: $silver-chalice;
  }

   &__img {
     width: 100%;
   }

   &__wrapper {
    @include flex($direction: column, $justify: center);
    height: 80%;
  }

  &__read-more {
    display: block;
	  padding: 0;
    margin: 20px 0 0;
    background: none;
	  border: none;
	  font: inherit;
    text-align: end;
    text-transform: uppercase;
	  color: inherit;
	  outline: inherit;
    cursor: pointer;
  
    &:hover {
      text-decoration: underline;
    }
  }
}

@media only screen and (min-width: 480px) {
  .blog {
    width: 100%;

    &:nth-of-type(2) {
      display: block;
    }

    &__title {
      font-size: 0.8rem;
    }
  
    &__description {
      font-size: 0.6rem;
    }

    &__wrapper {
      @include flex($wrap: nowrap);
    }
  }
}

@media only screen and (min-width: 768px) {
  .blog {
    &__title {
      font-size: 0.8rem;
    }
  
    &__description {
      font-size: 0.6rem;
    }
  }
}
