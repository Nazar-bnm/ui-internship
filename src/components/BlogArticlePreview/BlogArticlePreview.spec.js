import React from 'react';
import { shallow } from 'enzyme';
import BlogArticlePreview from './BlogArticlePreview';

describe('<BlogArticlePreview />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      imageSrc: 'src/assets/img/content/new2.png',
      heading: 'Raccoons are fun and fashionable',
      text: 'Raccoons are fun and fashionable, they like to eat vinegrapes',
    };

    wrapper = shallow(
      <BlogArticlePreview {...props} />
    );
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
