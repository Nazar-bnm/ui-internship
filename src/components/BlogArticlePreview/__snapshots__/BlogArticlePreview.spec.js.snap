// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogArticlePreview /> should match snapshot 1`] = `
<div
  className="article-preview"
  style={
    Object {
      "backgroundImage": "url(src/assets/img/content/new2.png)",
    }
  }
>
  <div
    className="article-preview__content"
  >
    <h2
      className="article-preview__heading"
    >
      Raccoons are fun and fashionable
    </h2>
    <p
      className="article-preview__text"
    >
      Raccoons are fun and fashionable, they like to eat vinegrapes
    </p>
  </div>
</div>
`;
